### ISSUE TITLE
All bookmarks are not sync'd when bookmark/folder are added, removed and re-added 

### ISSUE THREAD

**Issue by srirambv** [brave/brave-ios#985]
### Description: 
All bookmarks are not sync'd when bookmark/folder are added, removed and re-added 

### Devices
Device 1: iPad Pro (iOS 12.1.4) sync chain creator
Device 2: iPhone 7+ (iOS 12.1.4)

### Steps to Reproduce (reproduced using the steps 3 times)
  1. Create sync chain on device 1
  2. Add device 2 to the chain using QR code
  3. Add bookmarks (YT,Reddit,Wiki,Amazon) on device 1 and delete them before its sync'd to device 2
  4. Add the same bookmarks again and put one of the bookmark inside a folder
  5. Wait for the bookmarks to sync, all bookmarks are not sync'd 

**Actual result:** 
![image](https://user-images.githubusercontent.com/17010094/54546345-e26a5280-4979-11e9-96c8-6f24ad347e62.png)
![image](https://user-images.githubusercontent.com/17010094/54546354-e5654300-4979-11e9-8860-4a3872b20bfc.png)

Bookmarks on Device 1:
* Youtube 
* reddit 
* Amazon
* We (folder)
  * Wikipedia 

On Device 2, Wikipedia was sync'd twice and then removed

**Expected result:**
All bookmarks should be sync'd correctly 

**Reproduces how often:** [Easily reproduced, Intermittent Issue]
Fairly easy if you follow the steps correctly 

**Website problems only:**
- did you check with Brave Shields down?
- did you check in Safari/Firefox (WkWebView-based browsers)? 

### Additional Information 
On first attempt, there was no bookmark sync on device 2. When I added device 3 to the sync chain even that didnt get any bookmarks and didn't send any of the existing bookmarks to other two devices. On 2nd/3rd and 4th attempts i could reproduce the issue of missing bookmarks. 
Found the issue while testing #450
Adding P2 since this would be a real world scenario when a user adds a bookmark and removes it before its sync'd to the other devices and readds it

**Comment by srirambv** [comment]
Adding a new bookmark on device 1 doesn't propagate missing bookmarks or new one to other devices

**Comment by btlechowski** [comment]
I was unable to reproduce with
Device1: iPhone SE, Brave 1.9 (19.3.9.1) 
Device2: Windows 7, Brave 0.61.51 

The issue could be only reproducible when syncing 2 iOS devices.

**Comment by kjozwiak** [comment]
I ran into something similar when going through #450. I noticed that under the `macOS`, Amazon didn't get added into the correct folder and was directly placed into the bookmark toolbar, example:

![IMG_0224](https://user-images.githubusercontent.com/2602313/55167294-10545180-5147-11e9-9b72-1c147be5c9e7.PNG)

However, under `iOS/iPad`, the bookmark doesn't appear under the bookmark toolbar:

<img width="1046" alt="Screen Shot 2019-03-27 at 1 04 28 AM" src="https://user-images.githubusercontent.com/2602313/55167204-e1d67680-5146-11e9-917d-cfd80feed8ee.png">

I tried syncing a few times by adding new folders/moving things around but the Amazon bookmark on the desktop never changed and basically stayed on the toolbar. So I basically had two different instances of data on two devices that were being synced together.
